docs/design/interpreter/transformation.md

```diff
@@ -0,0 +1,618 @@
+* mono_ldstr_checked -> getStringLiteral
+* mono_marshal_get_managed_wrapper - N.A. for coreclr
+* mono_metadata_token_index -> constructStringLiteral or getLazyStringLiteralHelper
+  * Used in CEE_LDSTR
+* mono_metadata_token_table - N.A., these details are hidden behind token resolving
+  * Used in CEE_SIZEOF
+* mono_method_can_access_method -> getCallInfo, CORINFO_CALL_INFO::accessAllowed
+  * Used by CEE_LDFTN 
+* mono_method_get_context
+  * It is needed to get the instantiation argument 0 for Span and ReadOnlySpan constructor.
+  * So we can get it via getTypeInstantiationArgument(clsHandle, 0)
+* mono_method_get_wrapper_data - N.A., Mono specific
+* mono_method_has_unmanaged_callers_only_attribute -> N.A., it was used only for sanity checks and the the getCallInfo does those internally
+  * Used by CEE_LDFTN
+* mono_mint_type -> asCorInfoType
+* mono_trace_eval
+  * Mono specific, will need to have something similar for profiler for coreclr
+* mono_type_size(mono_type_create_from_typespec_checked) -> resolveToken, getClassSize( CORINFO_RESOLVED_TOKEN::hClass)
+  * Used in CEE_SIZEOF
+* mono_type_get_desc -> getClassNameFromMetadata
+* mono_type_get_full_name -> getClassNameFromMetadata
+* mono_type_get_object_checked -> getRuntimeTypePointer
+  * gets System.RuntimeType object for a MonoType
+  * Used by CEE_LDTOKEN
+
+## Functions that don't use any Mono APIs
+These functions might still use glib APIs for memory allocation, bitset, hashtable or linked list. Or they use mono_interp_* or mono_mint_* functions, which are local to the interpreter.
+
+* interp_new_ins
+* interp_add_ins
+* interp_add_ins_explicit
+* interp_insert_ins
+* interp_insert_ins_bb
+* interp_clear_ins
+* interp_ins_is_nop
+* interp_prev_ins
+* interp_next_ins
+* get_stack_size
+* get_tos_offset
+* interp_create_stack_var
+* set_type_and_var
+* set_simple_type_and_var
+* interp_make_var_renamable
+* interp_create_renamed_fixed_var
+* push_type
+* push_simple_type
+* push_type_vt
+* push_types
+* interp_get_mov_for_type
+* get_mint_type_size
+* try_fold_one_arg_branch
+* interp_add_conv
+* try_fold_two_arg_branch
+* unary_arith_op(TransformData *td, int mint_op)
+* can_store
+* emit_ldptr
+* mono_interp_print_td_code
+* interp_ip_in_cbb
+* interp_ins_is_ldc
+* get_type_comparison_op
+* check_stack_helper
+* ensure_stack
+* realloc_stack
+* push_type_explicit
+* fixup_newbb_stack_locals
+* init_bb_stack_state
+* one_arg_branch
+* store_local
+* init_last_ins_call
+* imethod_alloc0
+* interp_generate_icall_throw
+* interp_dump_compacted_ins
+* interp_dump_code
+* interp_dump_ins
+* get_data_item_wide_index
+* get_data_item_index
+* get_data_item_index_imethod
+* is_data_item_wide_index
+* get_data_item_index_nonshared
+* interp_dump_bb
+* interp_get_const_from_ldc_i4
+* interp_get_ldc_i4_from_const
+* interp_get_ldind_for_mt
+* interp_get_stind_for_mt
+* generate_compacted_code
+* mono_jiterp_insert_ins
```

Comment:
anything with 'jiterp' in the name can be safely ignored as it is specific to the wasm jiterpreter